scripts/generate_upload_requirements_layer.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+# This script is used to generate a dependency layer for the lambdas to utilize
+# It's not really for direct usage, `make generate-dependencies` can be used
+# but its part of the `make deploy` 
+
+set -euo pipefail
+
+if [[ -z $DSS_DEPLOYMENT_STAGE ]]; then
+    echo 'Please run "source environment" in the data-store repo root directory before running this command'
+    exit 1
+fi
+
+account_id=$(aws sts get-caller-identity | jq -r ".Account" )
+populated_dss_tf_backend=$(echo ${DSS_TERRAFORM_BACKEND_BUCKET_TEMPLATE} | sed 's/{account_id}/'"${account_id}"'/'  )
+
+if [[ $(aws s3api get-bucket-location --bucket ${populated_dss_tf_backend} &>/dev/null ;echo $?) -ne 0 ]]
+then
+	echo "verify if bucket: ${populated_dss_tf_backend} exist"
+	exit 1
+fi
+
+
+
+build_path="$DSS_HOME/dependencies/python/lib/python3.6/site-packages"
+dependency_dir="$DSS_HOME/dependencies"
+aws_req_key=$DSS_DEPLOYMENT_STAGE/requirements.txt
+aws_zip_key=$DSS_DEPLOYMENT_STAGE/dss-dependencies-$DSS_DEPLOYMENT_STAGE.zip
+local_req=$DSS_HOME/requirements.txt
+local_zip=$DSS_HOME/dss-dependencies-$DSS_DEPLOYMENT_STAGE.zip
+layer_name=dss-dependencies-${DSS_DEPLOYMENT_STAGE}
+
+function create_layer(){
+layer_version_arn=$(aws lambda publish-layer-version --layer-name $layer_name --content S3Bucket=${populated_dss_tf_backend},S3Key=${aws_zip_key}| jq -r .LayerVersionArn )
+echo "created layer/-version_arn: ${layer_version_arn}"
+}
+
+function check() {
+ aws_req_hash=$(aws s3api head-object --bucket ${populated_dss_tf_backend} --key ${aws_req_key} | jq -r .ETag | xxd -r -p | base64)
+ local_req_hash=$(openssl md5 -binary ${local_req} | base64)
+if [[ $aws_req_hash != $local_req_hash ]]; then
+ echo "checksum missmatch, uploading new ${aws_zip_key}"
+ upload
+else
+ echo "requirements have not changed, no need to update"
+ exit 0
+fi
+}
+
+function upload() {
```

Comment:
Please update the formatting in this file to be consistent with existing DSS bash scripting: 4 space indentation for functions and flow control.

Reply:
resolved